tools/mksurfdata_map/src/mkglcmecMod.F90

```diff
@@ -301,13 +305,16 @@ subroutine mkglcmec(ldomain, mapfname, &
         call abort()
      end if
 
+     ! Obtain frac_dst
+     call gridmap_calc_frac_dst(tgridmap, tdomain%mask, frac_dst)
+
      do n = 1,tgridmap%ns
         ni = tgridmap%src_indx(n)
         no = tgridmap%dst_indx(n)
-        wt = tgridmap%wovr(n)
+        wt = tgridmap%wovr(n) * tdomain%mask(ni)
 
         ! fraction of this destination cell that is covered by source cells that are within the source landmask
-        frac = tgridmap%frac_dst(no)
+        frac = frac_dst(no)
```

Comment:
One idea I have is this:

A few lines down from here we check
`if (frac > 0) then`
while in subr. gridmap_calc_frac_dst we check
`if (mask_src > 0) then`

I ran a test where the only thing I changed relative to this commit was to return to using old line 310 instead of new line 317, but that made no difference. I have **not** tried changing the if statement in subr. gridmap_calc_frac_dst...

Reply:
Add'l info... This is still true with latest commit:

new code vs. old code both pointing at masked SRC files gives
max(abs(rel(dif(ALL_VARIABLES)))) = 8.1e-13
where this value corresponds to PCT_NAT_PFT

Reply:
Also... where you can find my namelists:
/glade/work/slevis/git/mksurfdata_maps_wo_src_masks/tools/mksurfdata_map

Reply:
This was resolved [here.](https://github.com/ESCOMP/CTSM/pull/823#issuecomment-549932045)

Reply:
...and these notes are not otherwise helpful, so I will hit the resolved button.